### DISCUSSION TITLE
What kind of articles are required?

### DISCUSSION THREAD

**Discussion by hajimehoshi** [hajimehoshi/ebiten#1507] [General]
I'm thinking to write a book or articles to explain how to use Ebiten. I was wondering what kind of articles are on demand:

* Small examples and their explanations (like O'Reilly's cookbook series, or like "A tour of Go")
* Examples to develop small or medium size games (like Flappy or Blocks in Ebiten's examples directory)
* Examples to develop large size games (like Bear's Restaurant. I'd like to know how to do this btw!)
* How Ebiten works (implementation details)

Probably I'd make mixed ones, but I'd like to hear your opinion. Thanks,

**Comment by loig** [comment]
What I missed the most when I started using Ebiten were guidelines on how to use the different elements of the library (i.e. how this or that functionality was intended to be used). Something similar to what one can find in the "performance tips" part of your website. This is quite obvious (and well documented) for basic functionalities of the library (like how to implement the ebiten.Game interface) but it would be nice to have it also for basically the whole library.

Ideally, each such guideline would be accompanied by an example (or at least a reference to one of the examples available with Ebiten) and complemented by a short explanation on the implementation of the relevant part of Ebiten (explaining the technical details behind this guideline).

**Reply by hajimehoshi** [reply]
That's a good insight. Filling the gap between the API usage and implementing actual games is important!

**Reply by hajimehoshi** [reply]
Added an issue https://github.com/hajimehoshi/ebiten/issues/1528

**Comment by dcoty** [comment]
I think with the examples already provided, a code breakdown would be good. what's being done in this block of code and why. In other words, expand "How the code works" to include the games and other examples. 

This will definitely be more work ( and community involvement) but a video/blog series on developing small/medium/large games in Ebiten would be useful too. We could use more content like this to help folks make better sense of things: https://www.fatoldyeti.com/categories/roguelike-tutorial/

**Reply by hajimehoshi** [reply]
Adding more explanation to the existing examples sounds fine!

I have no idea how to make videos :-) Step-by-step tutorials to develop actual games would be nice, and this is the second item of my first ideas.

**Comment by jstaf** [comment]
I think it would be useful to create articles on individual tasks and small examples that cover common tasks and stuff that people would want - for example: how to create some common shaders that most people will want like FXAA, Bloom, etc.

**Reply by hajimehoshi** [reply]
The shader is still premature, and I understand we need more examples. Thanks!

**Comment by wasedaigo** [comment]
Small example usually does not cover how the packages should be structured in a real-world project. Also ebiten requires lots of primitive code in order to get started (example: asset loading). I'd think adding more functionalities to ebitenutils would make it much easier to write good samples.

**Comment by occamist** [comment]
1 and 2.

Small snippets with fully line by line commented examples make things easier. Because sometimes it is easy to get lost with godoc explanations

Small sized games like flappy bird, tic tac toe, mario, tetris, pacman, candy crush etc... is super great and welcoming for beginners and it doesn't need to be fully complete. It needs a line by line explanations

Existing samples are cool but I do think it is most important to give the basics right with existing ebiten functionality. Snippets also could be shared to show how fun and easy it is to use ebiten APIwise.

Examples/functional concepts could be extended like pixi.js or phaser.js examples, it doesn't have to be a game. In fact, I think writing a fully fledged big game would confuse a lot of beginners

I am personally aiming to give 1 networking game example with ebiten + gRPC
